Assignment_1/src/main/java/IPL/PlayerOperations.java

```
@@ -0,0 +1,208 @@
+        WicketKeeper keeper = new WicketKeeper(tokens.get(0), team, Integer.parseInt(tokens.get(3)),
+                Integer.parseInt(tokens.get(4)), Double.parseDouble(tokens.get(5)),
+                Double.parseDouble(tokens.get(6)), Integer.parseInt(tokens.get(7)));
+        keepers.add(keeper);
+        team.addWicketKeeper(keeper);
+    }
+
+    private static void addAllRounder(List<String> tokens, Team team) {
+        AllRounder allRounder = new AllRounder(tokens.get(0), team, Integer.parseInt(tokens.get(3)),
+                Integer.parseInt(tokens.get(4)), Double.parseDouble(tokens.get(5)),
+                Double.parseDouble(tokens.get(6)), Integer.parseInt(tokens.get(7)));
+        allRounders.add(allRounder);
+        team.addAllRounder(allRounder);
+    }
+
+    public static void searchPlayer() throws IOException {
+        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
+        System.out.println("Enter the name of the player : ");
+        String playerName = in.readLine();
+        if(playerName.equals("") || playerName.equals(" ")) {
+            System.out.println("Enter a valid name or part of a name.");
+            return;
+        }
+
+        List<Bowler> matchingBowlers = bowlers.stream().
+                filter(b -> b.getName().toLowerCase().contains(playerName.toLowerCase())).collect(Collectors.toList());
+        List<Batsman> matchingBatsmen = batsmen.stream().
+                filter(bt -> bt.getName().toLowerCase().contains(playerName.toLowerCase())).collect(Collectors.toList());
+        List<WicketKeeper> matchingWicketKeepers = keepers.stream().
+                filter(wk -> wk.getName().toLowerCase().contains(playerName.toLowerCase())).collect(Collectors.toList());
+        List<AllRounder> matchingAllRounders = allRounders.stream().
+                filter(ar -> ar.getName().toLowerCase().contains(playerName.toLowerCase())).collect(Collectors.toList());
```

Should raise an exception when the search result is empty